Mathlib/LinearAlgebra/QuadraticForm/Basic.lean

```diff
@@ -498,397 +512,435 @@ end SemiringOperators
+#align quadratic_form.associated_hom QuadraticMap.associatedHom
 
-variable (Q : QuadraticForm R M)
+variable (Q : QuadraticMap R M N)
 
 @[simp]
-theorem associated_apply (x y : M) : associatedHom S Q x y = ⅟ 2 * (Q (x + y) - Q x - Q y) :=
+theorem associated_apply (x y : M) : associatedHom S Q x y = ⅟ (2 : R) • (Q (x + y) - Q x - Q y) :=
   rfl
-#align quadratic_form.associated_apply QuadraticForm.associated_apply
+#align quadratic_form.associated_apply QuadraticMap.associated_apply
 
 @[simp] theorem two_nsmul_associated : 2 • associatedHom S Q = Q.polarBilin := by
   ext
   dsimp
-  rw [← smul_mul_assoc, two_nsmul, invOf_two_add_invOf_two, one_mul, polar]
+  rw [← smul_assoc, two_nsmul, invOf_two_add_invOf_two, one_smul, polar]
 
-theorem associated_isSymm : (associatedHom S Q).IsSymm := fun x y => by
+theorem associated_isSymm (Q : QuadraticMap R M R) : (associatedHom S Q).IsSymm := fun x y => by
   simp only [associated_apply, sub_eq_add_neg, add_assoc, map_mul, RingHom.id_apply, map_add,
     _root_.map_neg, add_comm, add_left_comm]
-#align quadratic_form.associated_is_symm QuadraticForm.associated_isSymm
+#align quadratic_form.associated_is_symm QuadraticMap.associated_isSymm
+
 
 @[simp]
-theorem associated_comp [AddCommGroup N] [Module R N] (f : N →ₗ[R] M) :
+theorem associated_comp {N' : Type*} [AddCommGroup N'] [Module R N'] (f : N' →ₗ[R] M) :
     associatedHom S (Q.comp f) = (associatedHom S Q).compl₁₂ f f := by
   ext
   simp only [associated_apply, comp_apply, map_add, LinearMap.compl₁₂_apply]
-#align quadratic_form.associated_comp QuadraticForm.associated_comp
+#align quadratic_form.associated_comp QuadraticMap.associated_comp
 
-theorem associated_toQuadraticForm (B : BilinForm R M) (x y : M) :
-    associatedHom S B.toQuadraticForm x y = ⅟ 2 * (B x y + B y x) := by
-  simp only [associated_apply, toQuadraticForm_apply, map_add, add_apply, ← polar_toQuadraticForm,
-    polar.eq_1]
-#align quadratic_form.associated_to_quadratic_form QuadraticForm.associated_toQuadraticForm
+theorem associated_toQuadraticMap (B : BilinMap R M R) (x y : M) :
+    associatedHom S B.toQuadraticMap x y = ⅟ (2 : R) • (B x y + B y x) := by
+  simp only [associated_apply, LinearMap.BilinMap.toQuadraticMap_apply, map_add,
+    LinearMap.add_apply, smul_eq_mul]
+  abel_nf
+#align quadratic_form.associated_to_quadratic_form QuadraticMap.associated_toQuadraticMap
 
-theorem associated_left_inverse (h : B₁.IsSymm) : associatedHom S B₁.toQuadraticForm = B₁ :=
-  LinearMap.ext₂ fun x y => by
-    rw [associated_toQuadraticForm, ← h.eq, RingHom.id_apply, ← two_mul, ← mul_assoc,
-      invOf_mul_self, one_mul]
-#align quadratic_form.associated_left_inverse QuadraticForm.associated_left_inverse
+theorem associated_left_inverse (h : B₁.IsSymm) : associatedHom S B₁.toQuadraticMap = B₁ :=
+    LinearMap.ext₂ fun x y => by
+  rw [associated_toQuadraticMap, ← h.eq x y, RingHom.id_apply, ← two_mul, ← smul_mul_assoc,
+    smul_eq_mul, invOf_mul_self, one_mul]
```

Comment:
```suggestion
  LinearMap.ext₂ fun x y => by
    rw [associated_toQuadraticMap, ← h.eq x y, RingHom.id_apply, ← two_mul, ← smul_mul_assoc,
      smul_eq_mul, invOf_mul_self, one_mul]
```
Note that this is the proof term, not part of the statement, so indent by 2 instead of 4.